### DISCUSSION TITLE
Emacs mode for V

### DISCUSSION THREAD

**Discussion by nordlow** [vlang/v#11622] [Ideas]
Has anybody put together an Emacs mode for V?

If not, I guess `go-mode` would be a good starting point for branching into `v-mode`.

**Comment by spytheman** [comment]
I do use `go-mode` for emacs in tandem with `smart-compile` , and that combination works very nicely.

Other changes related to V, that I've made to my .emacs are these lines:
```emacs
(require 'smart-compile)
(defun save-compile-buffer()
  (interactive)
  (save-buffer)
  (smart-compile 1)
)
(global-set-key [f5]   'save-compile-buffer)
(global-set-key [f6]   'next-error)
(add-to-list 'auto-mode-alist '("\\.v\\'" . go-mode))
(add-to-list 'smart-compile-alist
           '("\\.v\\'" . "VERROR_PATHS=absolute /usr/local/bin/v -g run %f"))
```

**Comment by elimisteve** [comment]
@spytheman Do you still do the above or do you use https://github.com/damon-kwok/v-mode ? Thanks!
